### ISSUE TITLE
failure to send if “Send As” > 15 characters

### ISSUE THREAD

**Issue by ncolton** [jlewallen/jenkins-hipchat-plugin#58]
Messages fail to be sent if the “Send As” value is greater than 15 characters:

```
Jun 04, 2014 4:31:10 PM WARNING jenkins.plugins.hipchat.StandardHipChatService publish
HipChat post may have failed. Response: {"error":{"code":400,"type":"Bad Request","message":"From name must be between 1 and 15 characters."}}
```

**Comment by toddpi314** [comment]
@ncolton, take a better look at the error. "From name" is the " Send As" field in the global plugin configuration. 

Maybe you can try shortening that value?
Also note, this is a REST error from Atlassian. The plugin just relays what you send it.

**Comment by toddpi314** [comment]
I tested this on the last 4 versions of Jenkins...
For some versions, you need to wrap the "Send As" value with quotes.

**Comment by killuazhu** [comment]
After I updated the " Send As" field in the global plugin configuration, although UI shows it's updated, from jenkins log hipchat plugin is still using old 'Send As' name.  Even jenkins restart won't update the value. 

I have to remove hipchat plugin, delete jenkins.plugins.hipchat.HipChatNotifier.xml, re-install hipchat plugin again to make it take effect.

**Comment by ncolton** [comment]
> @ncolton, take a better look at the error. "From name" is the " Send As" field in the global plugin configuration.
> 
> Maybe you can try shortening that value?
> Also note, this is a REST error from Atlassian. The plugin just relays what you send it.

Yes. I understand that. I failed to communicate that the issue for the plug-in is that it does not note any size restriction, does not limit the length of the Send As value that is settable, and does not make it obvious / clear that a Send As value beyond a certain character limit is a problem, and is the reason that messages aren't being sent through HipChat.

> I tested this on the last 4 versions of Jenkins...
> For some versions, you need to wrap the "Send As" value with quotes.

Does this mean that longer names are now useable? Not quite sure what you are saying here.

Sorry my response took so long!

**Comment by thom8** [comment]
This is the limit set on the Hipchat API -- https://www.hipchat.com/docs/api/method/rooms/message

"Name the message will appear be sent from. Must be less than 15 characters long. May contain letters, numbers, -, _, and spaces."

**Comment by jorgeorpinel** [comment]
Having this same problem, and like @killuazhu mentioned, Send As refuses to change according to the log althouogh the config page and xml files are updated. I tried his method or deleting xml and re-installing plugin to no avail. Stuck w/o HC notifications at the moment, I hope someone can fix this bugish.

**Comment by ncolton** [comment]
Perhaps it may help to clarify that my issue is _not_ with the behavior, but instead with the log message being less helpful than it could be.

Without knowing that the “Send As” configuration field is the “From name” the error message refers to, solving the problem is very difficult.

No indication in the user interface as to the limit, and the user interface allowing the setting of a “Send As” value longer than that permitted by the HipChat API, by lacking information and protection against this, increases the likelihood that users will encounter this problem.
